packages/actionmenu/src/react/item.tsx

```diff
@@ -43,99 +47,107 @@ const styles = {
   textOnly: () => css(stylesheet['.psds-actionmenu__text-only'])
 }
 
-interface Props extends Omit<HTMLAttributes<HTMLLIElement>, 'onClick'> {
-  disabled?: boolean
-  tagName?: string
+interface ItemProps extends Omit<HTMLAttributes<HTMLLIElement>, 'onClick'> {
   className?: string
-  origin?: string
-  onClick?: (event: React.MouseEvent, value: string | number) => void
+  disabled?: boolean
+  href?: string
   nested?: React.ReactNode
+  onClick?: (event: React.MouseEvent, value: string | number) => void
+  origin?: string
+  tagName?: string
   value?: string | number
 }
-
-export const Item = forwardRef<HTMLLIElement, Props>(
-  (
-    {
+export const Item = forwardRef<HTMLLIElement, ItemProps>(
+  (props, forwardedRef) => {
+    const {
+      children,
+      className,
       disabled,
       nested,
+      onClick,
       origin,
+      tagName: TagName = tagName.a,
       value,
-      className,
-      children,
-      onClick,
-      tagName,
       ...rest
-    },
-    forwardedRef
-  ) => {
-    const { onClickContext, onClose, originContext } = useContext<{
-      onClickContext: (e: Event, v: string | number) => void
-      onClose: (e: Event, v: string | number) => void
-      originContext: string
-    }>(ActionMenuContext)
+    } = props
+
+    const { onClickContext, onClose, originContext } = useContext(
+      ActionMenuContext
+    )
+
     const ref = useRef()
-    const subMenuRef = useRef()
     useImperativeHandle(forwardedRef, () => ref.current)
-    const TagName = tagName
+
+    const subMenuRef = useRef()
+
     const [open, setOpen] = useState(false)
     const hasSubMenu = Boolean(nested)
-    const handleMouseOver = e => {
+
+    const handleMouseOver: MouseEventHandler<HTMLLIElement> = () => {
```

Comment:
i just provided some more elaborate typings to this events to match the props...also i used the term `evt` instead of `e`. `evt` makes the clean coder part of my brain happier...

Reply:
Yeah, I'm usually `evt` too. To really Uncle Bob this though, should probably be `mouseoverEvent` or something. :)

Reply:
well that's a bridge to far... 😄